lib/widgets/message_list.dart

```diff
@@ -297,17 +297,26 @@ class MessageListAppBarTitle extends StatelessWidget {
   Widget _buildStreamRow(BuildContext context, {
     ZulipStream? stream,
   }) {
+    final store = PerAccountStoreWidget.of(context);
     final zulipLocalizations = ZulipLocalizations.of(context);
+
     // A null [Icon.icon] makes a blank space.
-    final icon = stream != null ? iconDataForStream(stream) : null;
+    IconData? icon;
+    Color? iconColor;
+    if (stream != null) {
+      icon = iconDataForStream(stream);
+      iconColor = colorSwatchFor(context, store.subscriptions[stream.streamId])
+        .iconOnBarBackground;
```

Comment:
Should we fall back to the default base color `kDefaultChannelColorSwatchBaseColor` (by passing a possibly null subscription) until we get to fully implementing #188?

Reply:
I think that's what this code already does; am I reading too quickly? `store.subscriptions[stream.streamId]` might be null, and if it is, `colorSwatchFor` uses a swatch based on `kDefaultChannelColorSwatchBaseColor`.

Reply:
Oh right. I think I meant the case when `stream` is `null`, i.e. we got a "(unknown channel)".